sdk/go/common/apitype/deployments.go

```diff
@@ -73,6 +73,20 @@ type CreateDeploymentRequest struct {
 
 	// Operation defines the options that the executor will use to run the Pulumi commands.
 	Operation *OperationContext `json:"operationContext,omitempty"`
+
+	AgentPoolID *AgentPoolIDMarshaller `json:"agentPoolID,omitempty"`
```

Comment:
@tgummerer I added this piece to handle pool id mashalling (see util_remote usage)